wireapp/wire-android
# feat: enable countly performance metrics (WPB-8842)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8842" title="WPB-8842" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-8842</a>  [Android] Implement Performance metrics provided by countly
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Enable performance analytics for the application

### Causes (Optional)

If the user has analytics enabled, the SDK will track some performance metrics.

### Solutions

What's tracked ?

- App startup, time in foreground, time on foreground
- Views names and general info about it (Not performance)

What's NOT tracked ?

- Networking, for now, due to layer not available for Countly
- Views, the SDK supports only the old Android View system, not compose.

### Testing

#### Test Coverage (Optional)


#### How to Test

Open, the app and play with the views and, you should see on countly tracking of views, and perf metrics for app startup foreground time, and background time,

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
